### DISCUSSION TITLE
Indicate when a long running search process is completed + clear previous results while running

### DISCUSSION THREAD

**Discussion by raindropsfromsky** [Flow-Launcher/Flow.Launcher#1643] [Ideas]
I have connected four external HDDs to my laptop. 
Thus, Flow has to handle a total of six volumes, including two logical partitions in my internal HDD.

I download a lot of files individually, rename them and move them between the HDDs frequently.

To avoid duplicates, I regularly use Flow to check whether any file is already on my HDDs.
At such times, I often see a blue highlight on the bottom edge of the Flow search bar.
(When Flow is waiting for the updation of the index, this highlight runs from left to right.)

While this design is attractive, it is functionally problematic: 
There is a small time interval between consecutive runs of the highlight.
As a result, it does not let you know instantly when the indexing is over.

There is a cognitive delay, which is a time-waster.
The user has to deliberately slow down to avoid rushing to a false negative conclusion.

It would be better to use the background color of the search bar to indicate the indexing status:

- As long as the bar is yellow/orange, the indexing is not over. The user should wait.
- As soon as the bar turns black, it means Flow is ready with the latest results.

With this design, Flow will instantly tell the user when the indexing is over.
The user will be able to avoid a false negative without watching the display closely.

**Comment by jjw24** [comment]
>I regularly use Flow to check whether any file is already on my HDDs.

Hi there, how are you checking the file, via searching the file name or searching with the path of the containing folder?

**Reply by raindropsfromsky** [reply]
Copying only the file base name from an internet source (no path, no extension), and then pasting it in Flow's search bar.

**Comment by jjw24** [comment]
Can you let us know how many files and average file size if possible when you often see the build highlight (progress bar), so we can replicate the exact scenario easier.

**Reply by raindropsfromsky** [reply]
These are mainly pdf/epub files, and their sizes vary from a few kB to a few MBs.

**Comment by raindropsfromsky** [comment]
Sorry about the delay in answering! 
(My bad- I had not turned on the GitHub control "watch for discussions")

**Comment by jjw24** [comment]
>As a result, it does not let you know instantly when the indexing is over.

Just a FYI in case you are not aware, we don't have control over the actual indexing of results, this is done by Everything plugin's indexer service or the Windows Search indexer. Explorer simply create a query against their API, I think when you say indexing status you actually mean the query results status? If so, let me know I will update the title and description as it is a bit misleading.

**Reply by raindropsfromsky** [reply]
> Just a FYI in case you are not aware, we don't have control over the actual indexing of results, this is done by Everything plugin's indexer service or the Windows Search indexer. Explorer simply create a query against their API, 

Yes, I am aware of this (someone described this logic in another issue I had filed earlier).

> I think when you say indexing status you actually mean the query results status? If so, let me know I will update the title and description as it is a bit misleading.

While the indexing is going on, FL's own bar shows a blue accent running left-to-right while it waits.
(to be precise, it used to be like that. I haven't observed that same behavior in the latest 1.11.0 version.)

So my point was, instead of showing the "wait" status with a dynamically running accent, just change the color of the bar.
For example, when we are supposed to wait for the index to get updated, show the bar in amber/orange.
When the results are ready, change the bar's color to the selected theme color.
That will instantly show when the wait is over.

**Comment by jjw24** [comment]
> So my point was, instead of showing the "wait" status with a dynamically running accent, just change the color of the bar.
For example, when we are supposed to wait for the index to get updated, show the bar in amber/orange.
When the results are ready, change the bar's color to the selected theme color.
That will instantly show when the wait is over.

Current design is where the blue accent bar stops when the search query is completed, I suppose this is not sufficient to indicate completion?

**Reply by raindropsfromsky** [reply]
> Current design is where the blue accent bar stops when the search query is completed, I suppose this is not sufficient to indicate completion?

Exactly!

It is difficult to spot just when the movement of the dark blue accent stops, as shown below:

![image](https://user-images.githubusercontent.com/9047168/213077353-799b53f8-9af7-447f-994e-03b44ddec60f.png)

BTW I forgot to mention another bug that's related: While the accent is running, FL keeps showing the _previous_ search results below the search bar. The user has to be aware that the results are NOT related to his latest search, and so he has to consciously avoid looking at the results while the accent is running. 

This conflict of information must be avoided by clearing off the previous results as soon as a new search is launched. Even if the new search has not yet yielded any results, the user must not be shown the previous results! 

In fact, in the worst case, the new search may not have any matching results. 
In such a case, when will FL clear off the previous results??

To sum up, we need **two** changes in the GUI:
1. Instead of a running accent, change the color of the search bar to indicate that the search is still running.
2. As soon as a new search is launched, FL must clear off the previous results.
